oidc-service-impl/src/main/java/io/mosip/esignet/services/AuthorizationServiceImpl.java

```diff
@@ -631,4 +641,35 @@ private void handleIdTokenHint(OAuthDetailRequestV3 oauthDetailReqDto, HttpServl
         }
     }
 
+    /**
+     * Set additional config in OIDC transaction based on openid profile and client additional config
+     * @param oidcTransaction  {@link OIDCTransaction}
+     * @param clientDetailDto  {@link ClientDetail}
+     * @param features {@link List<String>}
+     */
+    private void setAdditionalConfigInOidcTransaction(OIDCTransaction oidcTransaction, ClientDetail clientDetailDto, List<String> features) {
+        boolean par = features.contains(FEATURE_PAR);
+        boolean dpop = features.contains(FEATURE_DPOP);
+        boolean jwe = features.contains(FEATURE_JWE);
+        boolean pkce = features.contains(FEATURE_PKCE);
+
+        boolean isFapi = FAPI2.equalsIgnoreCase(openidProfile);
+        boolean isNisdsp = NISDSP.equalsIgnoreCase(openidProfile);
+
+        if (openidProfile == null || NONE.equalsIgnoreCase(openidProfile)) {
+            oidcTransaction.setUserInfoResponseType(clientDetailDto.getAdditionalConfig(USERINFO_RESPONSE_TYPE,FEATURE_JWS));
+            oidcTransaction.setDpopBoundAccessToken(clientDetailDto.getAdditionalConfig(DPOP_BOUND_ACCESS_TOKENS, false));
+            oidcTransaction.setRequirePushedAuthorizationRequests(clientDetailDto.getAdditionalConfig(REQUIRE_PAR, false));
+            oidcTransaction.setRequirePKCE(clientDetailDto.getAdditionalConfig(REQUIRE_PKCE, false));
+        } else {
+            if (isFapi || isNisdsp) {
```

Comment:
why is this check required? profile is configured in the properties, based on the configured profile, required features are available. 
First we should apply client configuration, then override the flags based on the configured profile. if the feature is not present in the list, do nothing.